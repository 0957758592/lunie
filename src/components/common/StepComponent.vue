<template>
  <div class="stepItem">
    <div class="circle--container">
      <div class="circle--default" :class="isCircleActive">
        {{ stepNumber }}
      </div>
      <p class="text--default" :class="isActive">
        {{ stepTitle }}
      </p>
    </div>
  </div>
</template>

<script>
export default {
  name: "StepComponent",
  props: {
    stepTitle: String,
    step: String,
    stepName: String,
    stepNumber: String,
    includeLine: {
      type: Boolean,
      default: true
    }
  },
  computed: {
    isActive() {
      return {
        textActive: this.step === this.stepName
      }
    },
    isCircleActive() {
      return {
        active: this.step === this.stepName
      }
    }
  }
}
</script>

<style>
.stepItem {
  display: flex;
  flex-direction: row;
  justify-content: center;
  width: 33%;
}

.circle--container {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.circle--default {
  border: 0.1rem var(--grey) solid;
  border-radius: 1rem;
  fill: transparent;
  width: 1.2rem;
  margin: 0 auto;
  margin-bottom: 0.3rem;
  font-size: var(--xs);
}

.active {
  border-color: var(--tertiary);
}

svg {
  display: inline-block;
}

.text--default {
  color: var(--grey);
  font-size: var(--xs);
  font-weight: 500;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  margin: 0 auto;
}

.grey--line {
  height: 1px;
  width: 20px;
  border: solid var(--grey) 1px;
}

.textActive {
  color: var(--tertiary);
}
</style>
