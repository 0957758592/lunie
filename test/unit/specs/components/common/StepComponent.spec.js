import StepComponent from "common/StepComponent"
import { shallowMount } from "@vue/test-utils"

describe(`TmBtn`, () => {
  let wrapper
  beforeEach(async () => {
    const value = `button`
    wrapper = shallowMount(StepComponent, {
      propsData: { value }
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`returns active class`, () => {
    wrapper.setProps({
      step: `Details`,
      stepName: `Details`
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`returns textActive class`, () => {
    wrapper.setProps({
      step: `Details`,
      stepName: `Details`
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
