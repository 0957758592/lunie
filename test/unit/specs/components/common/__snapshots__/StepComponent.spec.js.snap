// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmBtn has the expected html structure 1`] = `
<div
  class="stepItem"
  value="button"
>
  <svg
    height="30"
    width="30"
  >
    <circle
      class="circle--default active textActive"
      cx="15"
      cy="15"
      r="14"
    />
     
    <text
      dy=".3em"
      fill="white"
      stroke="white"
      stroke-width="1"
      text-anchor="middle"
      x="50%"
      y="50%"
    >
      
      
    
    </text>
  </svg>
  <!---->
   
  <p
    class="text--default active textActive"
  >
    
    
  
  </p>
</div>
`;

exports[`TmBtn returns active class 1`] = `
<div
  class="stepItem"
  value="button"
>
  <svg
    height="30"
    width="30"
  >
    <circle
      class="circle--default active textActive"
      cx="15"
      cy="15"
      r="14"
    />
     
    <text
      dy=".3em"
      fill="white"
      stroke="white"
      stroke-width="1"
      text-anchor="middle"
      x="50%"
      y="50%"
    >
      
      
    
    </text>
  </svg>
  <!---->
   
  <p
    class="text--default active textActive"
  >
    
    
  
  </p>
</div>
`;

exports[`TmBtn returns textActive class 1`] = `
<div
  class="stepItem"
  value="button"
>
  <svg
    height="30"
    width="30"
  >
    <circle
      class="circle--default active textActive"
      cx="15"
      cy="15"
      r="14"
    />
     
    <text
      dy=".3em"
      fill="white"
      stroke="white"
      stroke-width="1"
      text-anchor="middle"
      x="50%"
      y="50%"
    >
      
      
    
    </text>
  </svg>
  <!---->
   
  <p
    class="text--default active textActive"
  >
    
    
  
  </p>
</div>
`;
